angular/src/components/set-password.component.ts

```diff
@@ -69,30 +82,51 @@ export class SetPasswordComponent extends BaseChangePasswordComponent {
 
     async performSubmitActions(masterPasswordHash: string, key: SymmetricCryptoKey,
         encKey: [SymmetricCryptoKey, EncString]) {
+        // Create Set Password Request
         const request = new SetPasswordRequest();
         request.masterPasswordHash = masterPasswordHash;
         request.key = encKey[1].encryptedString;
         request.masterPasswordHint = this.hint;
         request.kdf = this.kdf;
         request.kdfIterations = this.kdfIterations;
         request.orgIdentifier = this.identifier;
-
+        // Create Key Pair
         const keys = await this.cryptoService.makeKeyPair(encKey[0]);
         request.keys = new KeysRequest(keys[0], keys[1].encryptedString);
 
+        // Make API call(s)
         try {
-            this.formPromise = this.apiService.setPassword(request);
-            await this.formPromise;
-
-            await this.userService.setInformation(await this.userService.getUserId(), await this.userService.getEmail(),
-                this.kdf, this.kdfIterations);
-            await this.cryptoService.setKey(key);
-            await this.cryptoService.setEncKey(encKey[1].encryptedString);
-            await this.cryptoService.setEncPrivateKey(keys[1].encryptedString);
+            if (this.resetPasswordAutoEnroll) {
+                this.formPromise = this.apiService.setPassword(request).then(async () => {
+                    await this.onSetPasswordSuccess(key, encKey, keys);
+                    return this.apiService.getOrganizationKeys(this.orgId);
+                }).then(async response => {
+                    if (response == null) {
+                        throw new Error(this.i18nService.t('resetPasswordOrgKeysError'));
+                    }
+
+                    const publicKey = Utils.fromB64ToArray(response.publicKey);
+
+                    // RSA Encrypt user's encKey.key with organization public key
+                    const userEncKey = await this.cryptoService.getEncKey();
+                    const encryptedKey = await this.cryptoService.rsaEncrypt(userEncKey.key, publicKey.buffer);
+
+                    // Create request and execute enrollment
+                    const resetRequest = new OrganizationUserResetPasswordEnrollmentRequest();
+                    resetRequest.resetPasswordKey = encryptedKey.encryptedString;
+
+                    // Get User Id
+                    const userId = await this.userService.getUserId();
```

Comment:
Most of the rest of these I would just remove, since they describe what the code is doing more than why it is doing it. Feel free to extract anything else that makes sense and improves readability, but we shouldn't write comments like these in my opinion. To quote Uncle Bob:

> The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures.

(I think he goes a little too crazy with it here but share the sentiment)

Reply:
Removed comments except for the one explaining the crypto operation.